pkg/cmd/tree.go

```diff
@@ -0,0 +1,88 @@
+package cmd
+
+import (
+	"fmt"
+
+	localworkflows "github.com/snyk/go-application-framework/pkg/local_workflows"
+	"github.com/snyk/go-application-framework/pkg/workflow"
+	"github.com/spf13/cobra"
+)
+
+type cmdTree struct {
```

Comment:
I may be blind... but is this used anywhere? 

Also, I don't think the command tree would belong into the go application framework - commands are only used in CLI applications, and go application framework is used as a library, not a CLI.

Reply:
Yes - this struct used here in the `Dev()` function: https://github.com/snyk/go-application-framework/pull/53/files#diff-f9ae7a73af8f4be90839b2b2fd48a2659dd25f5a4d157cfc86973247cb5b6069R28-R35

> Also, I don't think the command tree would belong into the go application framework - commands are only used in CLI applications, and go application framework is used as a library, not a CLI.

Just in case it's not clear - this package that I've added is a library for building a lightweight CLI, not a CLI itself.

This is something that I added to the extension that my team is developing in order to have a way to run our extension outside of the main Snyk CLI. It seems to me that the only alternative to get that kind of fast turnaround is to use a local `replace` directive in the cliv2 `go.mod`. Personally, I don't like to keep those around for very long, because you have to be careful not to commit them and, because `go.mod` and `go.sum` change pretty regularly during normal development, you need to stash it every time you pull in upstream changes. There's also lots of other stuff that the Snyk CLI does, for example analytics, that I don't really want when I just want to try out our extension.

I found this tool useful, so if I were going to write another extension, I would copy this code into it. That indicates that it would make sense as a library. But, if you don't think other extension authors would find it useful then maybe it doesn't belong in this library.

Reply:
Personally I think we are okay with this for now, for now it solves the purpose and since nothing is exported it should be relatively safe to change later on if necessary. Originally I was struggling a bit if this should be in `pkg` as this is part of the public API of the library. 

The  only thing in my mind is if we should call the package `dev` oder `devtools`, to be super even double explicit about its intention :)

Reply:
> The only thing in my mind is if we should call the package dev oder devtools, to be super even double explicit about its intention

`devtools.Cmd()` sounds pretty good to me. I'll push this change.

Reply:
Renamed in 7472c8e